packages/block-library/src/bookmark/index.php

```diff
@@ -0,0 +1,170 @@
+	switch ( $icon_type ) {
+		case 'bookmark':
+			$icon = $bookmark_icon;
+			break;
+		case 'heart':
+			$icon = $heart_icon;
+			break;
+		case 'star':
+			$icon = $star_icon;
+			break;
+		default:
+			$icon = $bookmark_icon;
+			break;
+	}
+
+	wp_interactivity_state(
+		'core/bookmark',
+		array(
+			'isBookmarked' => false,
+
+		)
+	);
+
+	$context_data = wp_interactivity_data_wp_context( $context );
+	wp_register_script_module( 'core-bookmark-view', '/wp-content/plugins/gutenberg/packages/block-library/src/bookmark/view.js' );
+	wp_enqueue_script_module( 'core-bookmark-view' );
+
+	if ( is_user_logged_in() ) {
+		$current_user_id = get_current_user_id();
+		$liked_posts     = array_map( 'intval', get_user_meta( $current_user_id, 'liked_posts', true ) ?? array() );
+		// Register a separate script to store user data.
+		wp_register_script(
+			'core-bookmark-user-data',
+			'/wp-content/plugins/gutenberg/packages/block-library/src/bookmark/bookmark-user-data.js',
+			array( 'wp-data', 'wp-core-data' ),
+			filemtime( '/wp-content/plugins/gutenberg/packages/block-library/src/bookmark/bookmark-user-data.js' ),
+			true
+		);
+
+		// Localize data to be available in `bookmark-user-data.js`
+		wp_localize_script(
+			'core-bookmark-user-data',
+			'bookmarkUserData',
+			array(
+				'userId'         => $current_user_id,
+				'userLikedPosts' => $liked_posts,
+			)
+		);
+
+		wp_enqueue_script( 'core-bookmark-user-data' );
+	}
+	return sprintf(
+		"<div
+		data-wp-interactive='core/bookmark'
+		%s
+		%s
+	>
+		<div
+			data-wp-on--click='actions.toggleBookmark'
+			data-wp-class--bookmark-active='state.isBookmarked'
+		>
+		<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' width='24' height='24' aria-hidden='true' focusable='false' fill='none' stroke='currentColor' stroke-width='2'><path d='%s'></path></svg>
+		</div>
+	</div>
+	",
+		$wrapper_attributes,
+		$context_data,
+		$icon
+	);
+}
+
+/**
+ * Registers the `bookmark` block, new user meta & rewrite rules.
+ *
+ * @since 6.8.0
+ */
+function register_block_core_bookmark() {
+	register_meta(
+		'user',
+		'liked_posts',
+		array(
+			'type'              => 'array',
+			'single'            => true,
+			'show_in_rest'      => array(
+				'schema' => array(
+					'type'  => 'array',
+					'items' => array( 'type' => 'integer' ),
+				),
+			),
+			'sanitize_callback' => function ( $value ) {
+				return is_array( $value ) ? array_map( 'sanitize_text_field', $value ) : array();
+			},
+			'auth_callback'     => function ( $allowed, $meta_key, $user_id ) {
+				return get_current_user_id() === (int) $user_id;
+			},
+		)
+	);
+
+	register_block_type_from_metadata(
+		__DIR__ . '/bookmark',
+		array(
+			'render_callback' => 'render_block_core_bookmark',
+		)
+	);
+
+	add_rewrite_rule( '^favorites/?$', 'index.php?pagename=favorites', 'top' );
+	add_rewrite_rule( '^favorites/page/([0-9]+)/?$', 'index.php?pagename=favorites&paged=$matches[1]', 'top' );
+	flush_rewrite_rules();
```

Comment:
Block shouldn't register any rewrite rules or flush them. Calling `flush_rewrite_rules` on `init` is a bad practice.